tensorpack/tfutils/argscope.py

```diff
@@ -78,12 +81,46 @@ def wrapped_func(*args, **kwargs):
     return wrapped_func
 
 
-def enable_argscope_for_module(module):
+def input_output_shape_decorator(func):
+    """Decorator for function to support input-output-shape information
+
+    Example:
```

Comment:
Let's make the format the same as tensorpack.models

Reply:
I guess
```
conv0:    [None, 28, 28, 1] ->   [None, 28, 28, 32]
```
is easier to parse than
```
conv0: input [None, 28, 28, 1]
conv0: output [None, 28, 28, 32]
```

Reply:
Agree. In tensorpack.models, layers can be nested, layers can have >1 inputs or outputs. That's why the logs are like that.
I think this format is fine.